```
Starting package 2/2: openni2:x64-windows
Building package openni2[core]:x64-windows...
-- Downloading https://github.com/OpenNI/OpenNI2/archive/2.2-beta2.tar.gz...
-- Extracting source E:/openni2/downloads/OpenNI-OpenNI2-2.2-beta2.tar.gz
-- Using source at E:/openni2/buildtrees/openni2/src/2.2-beta2-a8c7829707
-- Applying patch upgrade_projects.patch
-- Applying patch inherit_from_parent_or_project_defaults.patch
-- Applying patch replace_environment_variable.patch
CMake Error at ports/openni2/portfile.cmake:25 (message):
  Could not find <cor.h>.  Ensure the NETFXSDK is installed.
Call Stack (most recent call first):
  scripts/ports.cmake:84 (include)


Error: Building package openni2:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openni2:x64-windows
  Vcpkg version: 2019.08.16-nohash

Additionally, attach any relevant sections from the log files above.
```

@JackBoosY: Yeah seems like it is working. 
Maybe move the check to the top before it is downloading the source.

--- a/ports/openni2/portfile.cmake
+++ b/ports/openni2/portfile.cmake
@@ -1,7 +1,14 @@
 include(vcpkg_common_functions)
+
 if (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     message(FATAL_ERROR "Error: UWP builds are currently not supported.")
 endif()
+
+find_path(COR_H_PATH cor.h)
+if(COR_H_PATH MATCHES "NOTFOUND")
+    message(FATAL_ERROR "Could not find <cor.h>. Ensure the NETFXSDK is installed.")
+endif()
+get_filename_component(NETFXSDK_PATH "${COR_H_PATH}/../.." ABSOLUTE)
 
 vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY ONLY_DYNAMIC_CRT)
 
@@ -11,20 +18,10 @@ vcpkg_from_github(
     REF 2.2-beta2
     SHA512 60a3a3043679f3069aea869e92dc5881328ce4393d4140ea8d089027321ac501ae27d283657214e2834d216d0d49bf4f29a4b3d3e43df27a6ed21f889cd0083f
     HEAD_REF master
-)
-
-vcpkg_apply_patches(
-    SOURCE_PATH ${SOURCE_PATH}
     PATCHES upgrade_projects.patch
             inherit_from_parent_or_project_defaults.patch
             replace_environment_variable.patch
 )
-
-find_path(COR_H_PATH cor.h)
-if(COR_H_PATH MATCHES "NOTFOUND")
-    message(FATAL_ERROR "Could not find <cor.h>. Ensure the NETFXSDK is installed.")
-endif()
-get_filename_component(NETFXSDK_PATH "${COR_H_PATH}/../.." ABSOLUTE)
 
 file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})
 file(MAKE_DIRECTORY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})

